[BUG] errors downloading specs

running the command `sbt "run -es XQ31"`

A fatal error was thrown starting the test runner for the first time 

```
Uncaught error from thread [17 Nov 2020 19:36:26,784 [XQTSRunnerSystem-akka.actor.default-dispatcher-11] INFO  (slf4j.scala [info]:159) - Parsed XQTS TestSet: work/QT3_1_0/fn/timezone-from-time.xml OK.
17 Nov 2020 19:36:26,784 [XQTSRunnerSystem-akka.actor.default-dispatcher-11] INFO  (slf4j.scala [info]:159) - Parsing XQTS TestSet: work/QT3_1_0/fn/uri-collection.xml...
XQTSRunnerSystem-akka.actor.default-dispatcher-10]: io/lacuna/bifurcan/IEntry, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[XQTSRunnerSystem]
java.lang.NoClassDefFoundError: io/lacuna/bifurcan/IEntry
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:756)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at sbt.internal.ManagedClassLoader.findClass(ManagedClassLoader.java:92)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at org.exist.xquery.functions.fn.FnModule.<clinit>(FnModule.java:245)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.exist.xquery.XQueryContext.instantiateModule(XQueryContext.java:1644)
	at org.exist.xquery.XQueryContext.loadDefaults(XQueryContext.java:3028)
	at org.exist.xquery.XQueryContext.<init>(XQueryContext.java:428)
	at org.exist.xquery.XQueryContext.<init>(XQueryContext.java:422)
	at org.exist.xqts.runner.ExistConnection.getCompiledQuery$1(ExistServer.scala:254)
	at org.exist.xqts.runner.ExistConnection.$anonfun$executeQuery$30(ExistServer.scala:289)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:103)
	at cats.effect.internals.IORunLoop$.restart(IORunLoop.scala:40)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1(IOBracket.scala:48)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1$adapted(IOBracket.scala:34)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1(IOAsync.scala:37)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1$adapted(IOAsync.scala:37)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1(IORunLoop.scala:315)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1$adapted(IORunLoop.scala:314)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:37)
	at cats.effect.IO.unsafeRunAsync(IO.scala:274)
	at cats.effect.internals.IOPlatform$.unsafeResync(IOPlatform.scala:39)
	at cats.effect.IO.unsafeRunTimed(IO.scala:342)
	at cats.effect.IO.unsafeRunSync(IO.scala:256)
	at org.exist.xqts.runner.ExistConnection.executeQuery(ExistServer.scala:325)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$9(TestCaseRunnerActor.scala:247)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$8(TestCaseRunnerActor.scala:246)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$7(TestCaseRunnerActor.scala:245)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$6(TestCaseRunnerActor.scala:244)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$5(TestCaseRunnerActor.scala:243)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.runTestCase(TestCaseRunnerActor.scala:242)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCaseWithExist$5(TestCaseRunnerActor.scala:206)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:103)
	at cats.effect.internals.IORunLoop$.restartCancelable(IORunLoop.scala:50)
	at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:100)
	at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:67)
	at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:35)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.super$startLoop(TrampolineEC.scala:90)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.$anonfun$startLoop$1(TrampolineEC.scala:90)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:90)
	at cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
	at cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:42)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:80)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:58)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:177)
	at cats.effect.internals.IORunLoop$.restart(IORunLoop.scala:40)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1(IOBracket.scala:48)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1$adapted(IOBracket.scala:34)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1(IOAsync.scala:37)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1$adapted(IOAsync.scala:37)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1(IORunLoop.scala:315)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1$adapted(IORunLoop.scala:314)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:37)
	at cats.effect.IO.unsafeRunAsync(IO.scala:274)
	at cats.effect.internals.IOPlatform$.unsafeResync(IOPlatform.scala:39)
	at cats.effect.IO.unsafeRunTimed(IO.scala:342)
	at cats.effect.IO.unsafeRunSync(IO.scala:256)
	at org.exist.xqts.runner.TestCaseRunnerActor.org$exist$xqts$runner$TestCaseRunnerActor$$runTestCaseWithExist(TestCaseRunnerActor.scala:215)
	at org.exist.xqts.runner.TestCaseRunnerActor$$anonfun$receive$1.applyOrElse(TestCaseRunnerActor.scala:189)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at org.exist.xqts.runner.TestCaseRunnerActor.aroundReceive(TestCaseRunnerActor.scala:59)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:577)
	at akka.actor.ActorCell.invoke(ActorCell.scala:547)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.ClassNotFoundException: io.lacuna.bifurcan.IEntry
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at sbt.internal.ManagedClassLoader.findClass(ManagedClassLoader.java:92)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 98 more
Uncaught error from thread [XQTSRunnerSystem-akka.actor.default-dispatcher-7]: Could not initialize class org.exist.xquery.functions.fn.FnModule, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[XQTSRunnerSystem]
java.lang.NoClassDefFoundError: Could not initialize class org.exist.xquery.functions.fn.FnModule
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.exist.xquery.XQueryContext.instantiateModule(XQueryContext.java:1644)
	at org.exist.xquery.XQueryContext.loadDefaults(XQueryContext.java:3028)
	at org.exist.xquery.XQueryContext.<init>(XQueryContext.java:428)
	at org.exist.xquery.XQueryContext.<init>(XQueryContext.java:422)
	at org.exist.xqts.runner.ExistConnection.getCompiledQuery$1(ExistServer.scala:254)
	at org.exist.xqts.runner.ExistConnection.$anonfun$executeQuery$30(ExistServer.scala:289)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:103)
	at cats.effect.internals.IORunLoop$.restart(IORunLoop.scala:40)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1(IOBracket.scala:48)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1$adapted(IOBracket.scala:34)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1(IOAsync.scala:37)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1$adapted(IOAsync.scala:37)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1(IORunLoop.scala:315)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1$adapted(IORunLoop.scala:314)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:37)
	at cats.effect.IO.unsafeRunAsync(IO.scala:274)
	at cats.effect.internals.IOPlatform$.unsafeResync(IOPlatform.scala:39)
	at cats.effect.IO.unsafeRunTimed(IO.scala:342)
	at cats.effect.IO.unsafeRunSync(IO.scala:256)
	at org.exist.xqts.runner.ExistConnection.executeQuery(ExistServer.scala:325)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$9(TestCaseRunnerActor.scala:247)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$8(TestCaseRunnerActor.scala:246)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$7(TestCaseRunnerActor.scala:245)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$6(TestCaseRunnerActor.scala:244)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCase$5(TestCaseRunnerActor.scala:243)
	at scalaz.$bslash$div.flatMap(Either.scala:141)
	at org.exist.xqts.runner.TestCaseRunnerActor.runTestCase(TestCaseRunnerActor.scala:242)
	at org.exist.xqts.runner.TestCaseRunnerActor.$anonfun$runTestCaseWithExist$5(TestCaseRunnerActor.scala:206)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:103)
	at cats.effect.internals.IORunLoop$.restartCancelable(IORunLoop.scala:50)
	at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:100)
	at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:67)
	at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:35)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.super$startLoop(TrampolineEC.scala:90)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.$anonfun$startLoop$1(TrampolineEC.scala:90)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:90)
	at cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
	at cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:42)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:80)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:58)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:177)
	at cats.effect.internals.IORunLoop$.restart(IORunLoop.scala:40)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1(IOBracket.scala:48)
	at cats.effect.internals.IOBracket$.$anonfun$apply$1$adapted(IOBracket.scala:34)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1(IOAsync.scala:37)
	at cats.effect.internals.IOAsync$.$anonfun$apply$1$adapted(IOAsync.scala:37)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1(IORunLoop.scala:315)
	at cats.effect.internals.IORunLoop$.$anonfun$suspendAsync$1$adapted(IORunLoop.scala:314)
	at cats.effect.internals.IORunLoop$RestartCallback.start(IORunLoop.scala:439)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:155)
	at cats.effect.internals.IORunLoop$.start(IORunLoop.scala:37)
	at cats.effect.IO.unsafeRunAsync(IO.scala:274)
	at cats.effect.internals.IOPlatform$.unsafeResync(IOPlatform.scala:39)
	at cats.effect.IO.unsafeRunTimed(IO.scala:342)
	at cats.effect.IO.unsafeRunSync(IO.scala:256)
	at org.exist.xqts.runner.TestCaseRunnerActor.org$exist$xqts$runner$TestCaseRunnerActor$$runTestCaseWithExist(TestCaseRunnerActor.scala:215)
	at org.exist.xqts.runner.TestCaseRunnerActor$$anonfun$receive$1.applyOrElse(TestCaseRunnerActor.scala:189)
	at akka.actor.Actor.aroundReceive(Actor.scala:537)
	at akka.actor.Actor.aroundReceive$(Actor.scala:535)
	at org.exist.xqts.runner.TestCaseRunnerActor.aroundReceive(TestCaseRunnerActor.scala:59)
	at akka.actor.ActorCell.receiveMessage(ActorCell.scala:577)
	at akka.actor.ActorCell.invoke(ActorCell.scala:547)
	at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
	at akka.dispatch.Mailbox.run(Mailbox.scala:231)
	at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
```

Okay @line-o and @joewiz , so I built a completely new and clean VM to ensure no problems. I then installed the following:

* OpenJDK 8 1.8.0_275
* Maven 3.6.3
* Scala  2.13.1
* SBT 1.4.3

I then performed the following steps:
```
mkdir code && cd code
git clone https://github.com/exist-db/exist.git
cd exist
mvn clean install -DskipTests

cd ..
git clone https://github.com/exist-db/exist-xqts-runner.git
cd exist-xqts-runner
sbt clean assembly
```

From there I then ran the assembly using the command `target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar -x HEAD`.
The `HEAD` part indicates that it should use the latest XQTS set of tests maintained on GitHub; the default otherwise is to use the XQTS 3.1 from the W3C which is a little aged and has a few bugs.

I did not see anyof the  errors that @line-o that look like: `Caused by: java.lang.ClassNotFoundException: io.lacuna.bifurcan.IEntry`.

The full XQTS test run ran to completion. I captured the console output and have attached the log for your reference - 
[HEAD.log](https://github.com/eXist-db/exist-xqts-runner/files/5566247/HEAD.log)

--- a/README.md
+++ b/README.md
@@ -9,6 +9,9 @@ This application executes a W3C XQTS against an embedded eXist-db server.
 
 
 ## Compiling from Source
+
+### Prerequisites
+
 To build from source you will need the following pre-requisites:
 
 1. Git Command Line tools.
@@ -18,11 +21,21 @@ To build from source you will need the following pre-requisites:
 
 In the following steps, we assume that git, java and sbt are available on your system path.
 
-The version of eXist-db that the XQTS driver is compiled for is set in `build.sbt`. If you wish to compile against a newer or custom version of eXist-db, you can modify this to the version of an eXist-db Maven/Ivy artifact which you have available to your system, e.g.:
+### Get the source code
+
+1. `git clone https://github.com/exist-db/exist-xqts-runner.git`
+2. `cd exist-xqts-runner`
+
+### Select target eXist-db version
+
+The version of eXist-db that the XQTS driver is compiled for is set in [build.sbt](build.sbt). 
+If you wish to compile against a newer or custom version of eXist-db, you must modify this to the version of an eXist-db Maven/Ivy artifact which you have available to your system, e.g.:
 
 ```scala
 val existV = "7.0.0-SNAPSHOT"
 ``` 
+
+exist-xqts-runner will check your local maven repository for a version matching the value of `existV`.
 
 If you set the version to a SNAPSHOT version you want to load from Github Maven Package repository you need to authenticate with the Github PAT.
 
@@ -30,14 +43,52 @@ It can be provided via a credentials file in ~/.ivy2/.credentials or by setting 
 
 Once the pre-requisites are met, to build from source you can execute the following commands from your console/terminal:
 
-1. `git clone https://github.com/exist-db/exist-xqts-runner.git`
-2. `cd exist-xqts-runner`
-3. `sbt compile`
+In your local checkout of eXistdb run:
+
+```bash
+mvn clean install -DskiptTests=true
+```
+
+This will add your local development version to the maven repository (usually `~/.m2`).
+
+### Packaging the Application from Compiled Source
+
+Create a standalone application (also known as an Uber Jar, Assembly, etc.) with
+
+```bash
+sbt assembly
+```
+
+You should now have it available at `target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar`. 
+
+**NOTE** If you require a standard Jar file for some purpose you can run `sbt package`, which will generate `target/scala-2.13/exist-xqts-runner_2.13-1.0.0.jar`.
+
+### Running the Packaged Application
+
+Given the standalone application, you can execute it by running either (on Linux/Mac):
+
+- `target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar`
+  as the executable header is compiled into the Jar file.
+- `java -jar exist-xqts-runner-assembly-1.0.0.jar` also works
+
+**NOTE:** It is recommended to run against the latest version of the testsuite with 
+
+```bash
+target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar -x HEAD
+```
+
+### Compiling
+
+Once the pre-requisites are met, to build from source execute the following commands from your console/terminal:
+
+```bash
+sbt compile
+```
 
 The compiled application is now available in the sub-directory `target/scala-2.13`.
 
-
 ### Running from Compiled Source
+
 If you wish to run the application from the compiled source code, you can run the following to display the arguments accepted by `exist-xqts-runner`:
 
 ```bash
@@ -46,22 +97,7 @@ sbt "run --help"
 
 Obviously you should study the output from `--help`, and make sure to set the command line arguments that you need.
 
-**NOTE**: When running `exist-xqts-runner` via. `sbt`, the `run` command and any subsequent arguments to `exist-xqts-runner` must all be enclosed in the same double-quotes.
-
-
-### Packaging the Application from Compiled Source
-* If you require a standard Jar file for some purpose you can run `sbt package`, which will generate `target/scala-2.13/exist-xqts-runner_2.13-1.0.0.jar`.
-
-* If you wish to create a standalone application (also known as an Uber Jar, Assembly, etc.) you can run `sbt assembly`, which will generate `target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar`. 
-
-
-### Running the Packaged Application
-Given the standalone application, you can execute it by running either:
-
-1. `java -jar exist-xqts-runner-assembly-1.0.0.jar`
-
-2. or, even by just executing the `exist-xqts-runner-assembly-1.0.0.jar` file directly, as we compile an executable header into the Jar file. e.g. (on Linux/Mac): `./exist-xqts-runner-assembly-1.0.0.jar`.
-
+**NOTE**: When running `exist-xqts-runner` via. `sbt`, the `run` command and any subsequent arguments to `exist-xqts-runner` must all be enclosed in the same double-quotes. If you want to execute the complete test suite, running the [packaged application](#Packaging-the-Application-from-Compiled-Source) is advised.
 
 ## Publishing 
 

--- a/README.md
+++ b/README.md
@@ -16,7 +16,7 @@ To build from source you will need the following pre-requisites:
 
 1. Git Command Line tools.
 2. Java 21
-3. SBT (Simple Build Tool) 1.10.11
+3. SBT (Simple Build Tool) 1.12.11 (the version pinned in [project/build.properties](project/build.properties))
 4. a Github personal access token (PAT) with public read access
 
 In the following steps, we assume that git, java and sbt are available on your system path.
@@ -46,7 +46,7 @@ Once the pre-requisites are met, to build from source you can execute the follow
 In your local checkout of eXistdb run:
 
 ```bash
-mvn clean install -DskiptTests=true
+mvn clean install -DskipTests=true
 ```
 
 This will add your local development version to the maven repository (usually `~/.m2`).
@@ -59,22 +59,23 @@ Create a standalone application (also known as an Uber Jar, Assembly, etc.) with
 sbt assembly
 ```
 
-You should now have it available at `target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar`. 
+You should now have it available at `target/scala-2.13/exist-xqts-runner-assembly-2.0.0-SNAPSHOT.jar`.
+The version in the file name is the one declared in [version.sbt](version.sbt), so adjust the paths below if it differs.
 
-**NOTE** If you require a standard Jar file for some purpose you can run `sbt package`, which will generate `target/scala-2.13/exist-xqts-runner_2.13-1.0.0.jar`.
+**NOTE** If you require a standard Jar file for some purpose you can run `sbt package`, which will generate `target/scala-2.13/exist-xqts-runner_2.13-2.0.0-SNAPSHOT.jar`.
 
 ### Running the Packaged Application
 
 Given the standalone application, you can execute it by running either (on Linux/Mac):
 
-- `target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar`
+- `target/scala-2.13/exist-xqts-runner-assembly-2.0.0-SNAPSHOT.jar`
   as the executable header is compiled into the Jar file.
-- `java -jar exist-xqts-runner-assembly-1.0.0.jar` also works
+- `java -jar exist-xqts-runner-assembly-2.0.0-SNAPSHOT.jar` also works
 
 **NOTE:** It is recommended to run against the latest version of the testsuite with 
 
 ```bash
-target/scala-2.13/exist-xqts-runner-assembly-1.0.0.jar -x HEAD
+target/scala-2.13/exist-xqts-runner-assembly-2.0.0-SNAPSHOT.jar -x HEAD
 ```
 
 ### Compiling
@@ -125,13 +126,19 @@ You can also publish to your local m2 repository. This is useful for testing new
 ## XQTS Results
 The results of executing the XQTS will be formatted as JUnit test output.
 
-* The JUnit report data will be written to the `target/junit/data` folder.
-* An HTML aggregate report summary will be written to `target/junit/html/index.html`.
+Everything is written below the output directory, which defaults to `target` and can be changed with `-o` (`--output-dir`):
+
+* The JUnit report data will be written to the `junit/data` sub-folder.
+* An HTML aggregate report summary will be written to `junit/html/index.html`.
+* A `runner-info.xml` file will be written to the root of the output directory. It records which runner produced the results, so that a report can be traced back to the code that generated it. It contains:
+  * `runner-jar` — the Git commit and build metadata compiled into the Jar's manifest, plus the SHA-256 checksum and path of the Jar itself.
+  * `embedded-exist-core` — the version of the `exist-core` artifact that the run was executed against.
+  * `run-info` — the start and completion timestamps, the XQTS version, and the number of test cases that were run.
 
 
 ## Application Architecture
-The application is constructed using [Akka](https://akka.io), and as such makes use of Actors to perform many tasks in parallel; This hopefully speeds up the process of running the XQTS which includes many test cases.
-When the Application first executes, it will check for a local copy of the XQTS in a sub-directory named `work`, if it cannot find a local copy of the XQTS then it will download it from the W3C.
+The application is constructed using [Apache Pekko](https://pekko.apache.org), and as such makes use of Actors to perform many tasks in parallel; This hopefully speeds up the process of running the XQTS which includes many test cases.
+When the Application first executes, it will check for a local copy of the XQTS in a sub-directory named `work` (configurable with `-l`, `--local-dir`), if it cannot find a local copy of the XQTS then it will download it. Version `3.1` is downloaded from the W3C at https://dev.w3.org/2011/QT3-test-suite/releases/QT3_1_0.zip, whereas `HEAD` — the recommended version — is downloaded from the W3C's [qt3tests](https://github.com/w3c/qt3tests) repository on Github. The download locations are configured in [application.conf](src/main/resources/application.conf).
 
 
 ### Actor Hierarchy
@@ -141,6 +148,6 @@ Actors naturally have a supervision hierarchy, where if an Actor fails it's supe
 
 
 ### Actor Message Flows
-Actors in the system communicate by sending messages between each other. The message flow of an Akka Actor system can sometimes be tricky to follow by reading the code. The diagram below attempts to inform the developer about the message flows in the system.
+Actors in the system communicate by sending messages between each other. The message flow of a Pekko Actor system can sometimes be tricky to follow by reading the code. The diagram below attempts to inform the developer about the message flows in the system.
 
 ![Actor Message Flow](https://github.com/exist-db/exist-xqts-runner/raw/main/doc/actor-message-flow.png "Actor Message Flow")

